modules/core/include/opencv2/core/mat.inl.hpp

```diff
@@ -584,6 +584,16 @@ Mat::Mat(const std::initializer_list<_Tp> list)
         return;
     Mat((int)list.size(), 1, traits::Type<_Tp>::value, (uchar*)list.begin()).copyTo(*this);
 }
+
+template<typename _Tp, typename> inline
+Mat::Mat(const std::initializer_list<int> sizes, const std::initializer_list<_Tp> list)
+    : flags(MAGIC_VAL | traits::Type<_Tp>::value | CV_MAT_CONT_FLAG), dims(2), rows((int)list.size()),
+      cols(1), data(0), datastart(0), dataend(0), datalimit(0), allocator(0), u(0), size(&rows), step(0)
+{
+    if(sizes.size() != 1 || *sizes.begin() == 0 || list.size() == 0)
+        return;
+    Mat((int)list.size() / *sizes.begin(), *sizes.begin(), traits::Type<_Tp>::value, (uchar*)list.begin()).copyTo(*this);
```

Comment:
`(int)list.size() / *sizes.begin()`

This is not "dims" parameter.
"dims" is `sizes.size()`